opensearch_py_ml/ml_commons/IAMRoleHelper.py

```diff
@@ -0,0 +1,301 @@
+            return True
+        except ClientError as e:
+            if e.response["Error"]["Code"] == "NoSuchEntity":
+                return False
+            else:
+                print(f"An error occurred: {e}")
+                return False
+
+    def delete_role(self, role_name):
+        """
+        Delete an IAM role along with its attached policies.
+
+        :param role_name: Name of the IAM role to delete.
+        """
+        iam_client = boto3.client("iam")
+
+        try:
+            # Detach managed policies from the role
+            policies = iam_client.list_attached_role_policies(RoleName=role_name)[
+                "AttachedPolicies"
+            ]
+            for policy in policies:
+                iam_client.detach_role_policy(
+                    RoleName=role_name, PolicyArn=policy["PolicyArn"]
+                )
+            print(f"All managed policies detached from role {role_name}.")
+
+            # Delete inline policies associated with the role
+            inline_policies = iam_client.list_role_policies(RoleName=role_name)[
+                "PolicyNames"
+            ]
+            for policy_name in inline_policies:
+                iam_client.delete_role_policy(
+                    RoleName=role_name, PolicyName=policy_name
+                )
+            print(f"All inline policies deleted from role {role_name}.")
+
+            # Finally, delete the IAM role
+            iam_client.delete_role(RoleName=role_name)
+            print(f"Role {role_name} deleted.")
+
+        except ClientError as e:
+            if e.response["Error"]["Code"] == "NoSuchEntity":
+                print(f"Role {role_name} does not exist.")
+            else:
+                print(f"An error occurred: {e}")
+
+    def create_iam_role(self, role_name, trust_policy_json, inline_policy_json):
+        """
+        Create a new IAM role with specified trust and inline policies.
+
+        :param role_name: Name of the IAM role to create.
+        :param trust_policy_json: Trust policy document in JSON format.
+        :param inline_policy_json: Inline policy document in JSON format.
+        :return: ARN of the created role or None if creation failed.
+        """
+        iam_client = boto3.client("iam")
+
+        try:
+            # Create the role with the provided trust policy
+            create_role_response = iam_client.create_role(
+                RoleName=role_name,
+                AssumeRolePolicyDocument=json.dumps(trust_policy_json),
+                Description="Role with custom trust and inline policies",
+            )
+
+            # Retrieve the ARN of the newly created role
+            role_arn = create_role_response["Role"]["Arn"]
+
+            # Attach the inline policy to the role
+            iam_client.put_role_policy(
+                RoleName=role_name,
+                PolicyName="InlinePolicy",  # Replace with preferred policy name if needed
```

Comment:
this can cause issue if a policy with the same name exists. 

1. We can take a input for policy name. 
2. if the policy name is not provided then we can generate a random policy name:

```
# Generate a default policy name if none is provided
        if policy_name is None:
            policy_name = f"InlinePolicy-{role_name}-{datetime.utcnow().strftime('%Y%m%d%H%M%S')}"
```